datanode/snapshot/service_create_snapshot.go

```diff
@@ -0,0 +1,257 @@
+package snapshot
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"fmt"
+	"hash"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"time"
+
+	"code.vegaprotocol.io/vega/datanode/metrics"
+	"code.vegaprotocol.io/vega/logging"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+type Meta struct {
+	ChainID                  string
+	HeightFrom               int64
+	HeightTo                 int64
+	CurrentStateSnapshotFile string
+	HistorySnapshotFile      string
+	WriteLockFile            string
```

Comment:
`WriteLockFile` is only used in a test. I think the test can work it out for itself based on what it passes into `CreateSnapshot`, and then we don't have this test-only field in production code.

Reply:
it wasn't being read any more, I've removed it.